src/Core/Authorization/AuthorizationResolver.cs

```diff
@@ -126,9 +128,10 @@ public bool IsStoredProcedureExecutionPermitted(string entityName, string roleNa
         }
 
         /// <inheritdoc />
-        public bool AreColumnsAllowedForOperation(string entityIdentifier, string roleName, EntityActionOperation operation, IEnumerable<string> columns)
+        public bool AreColumnsAllowedForOperation(string entityName, string roleName, EntityActionOperation operation, IEnumerable<string> columns)
         {
-            string entityName = _metadataProvider.GetEntityName(entityIdentifier);
+            string dataSourceName = _runtimeConfigProvider.GetConfig().GetDataSourceNameFromEntityName(entityName);
+            ISqlMetadataProvider metadataProvider = _metadataProviderFactory.GetMetadataProvider(dataSourceName);
```

Comment:
This looks like it may cause a regression with Cosmos' nested filter authorization. I'm not sure why this wasn't caught in an existing test, but the usage of `entityIdentifier` was intentional as it could represent a GraphQL Object Type name that was overridden by an entity's settings in the runtime config (GraphQL - Singular name if i remember correctly). 
`_metadataprovider.GetEntityName(entityIdentifier)` is resolving a graphQL object type name to an entity name.

Reply:
discussed offline and wherever there is a call to the function. I derive the entityName beforehand. (Found only one call where the graphqltypename is being sent and thats from the graphqlfilterparser. fixed that.